packages/encoding/src/decoding/decodeDexOutput.ts

```diff
@@ -0,0 +1,50 @@
+import { PedersenHash } from '@explorer/types'
+
+import { StarkExDexOutput } from '../OnChainData'
+import { ByteReader } from './ByteReader'
+
+export function decodeDexOutput(data: string): StarkExDexOutput {
+  const reader = new ByteReader(data)
+
+  const configCode = reader.readBigInt(32)
+  const initialValidiumVaultRoot = PedersenHash(reader.readHex(32))
+  const finalValidiumVaultRoot = PedersenHash(reader.readHex(32))
+  const initialRollupVaultRoot = PedersenHash(reader.readHex(32))
+  const finalRollupVaultRoot = PedersenHash(reader.readHex(32))
+  const initialOrderRoot = PedersenHash(reader.readHex(32))
+  const finalOrderRoot = PedersenHash(reader.readHex(32))
+  const globalExpirationTimestamp = reader.readBigInt(32)
+  const validiumVaultTreeHeight = reader.readNumber(32)
+  const rollupVaultTreeHeight = reader.readNumber(32)
+  const orderTreeHeight = reader.readNumber(32)
+  const nModifications = reader.readNumber(32)
+  const nConditionalTransfers = reader.readNumber(32)
+  const nL1VaultUpdates = reader.readNumber(32)
+  const nL1OrderMessages = reader.readNumber(32)
```

Comment:
instead of `nName` I'd use `nameCount` similarily to https://github.com/l2beat/starkex-explorer/blob/1525003d800f27596dd31ab81e44f79b778295da/packages/encoding/src/decoding/readFundingEntries.ts#L1

Also, what's the reason for `L1` prefix? maybe it can be dropped?

Reply:
The `L1` prefix comes straight from the Cairo code https://github.com/starkware-libs/starkex-for-spot-trading/blob/607f0b4ce507e1d95cd018d206a2797f6ba4aab4/src/starkware/cairo/dex/main.cairo#L21-L37

Reply:
@adamiak wdyt we should do about the naming here?

Reply:
Switching from `nX` to `XCount` is 👍 

As for the `L1*` fields, I'm not sure what they're for. So for the time being I would keep the names (but with lowercase 'l', i.e. `l1VaultUpdateCount` etc.)

@Wendrowiec13 when adding `...Count` please make the word root singular, e.g. it should be `modificationCount` and not `modificationsCount`. From what I checked (on the web and with @sz-piotr) this would be more grammatically correct.

Reply:
Done